AudioDeviceExample/AudioDevices/ExampleCoreAudioDevice.m

```diff
@@ -0,0 +1,487 @@
+//
+//  ExampleCoreAudioDevice.m
+//  AudioDeviceExample
+//
+//  Copyright © 2018 Twilio, Inc. All rights reserved.
+//
+
+#import "ExampleCoreAudioDevice.h"
+
+// We want to get as close to 10 msec buffers as possible because this is what the media engine prefers.
+static double const kPreferredIOBufferDuration = 0.01;
+// We will use stereo playback where available. Some audio routes may be restricted to mono only.
+static size_t const kPreferredNumberOfChannels = 2;
+static uint32_t const kPreferredSampleRate = 48000;
+
+typedef struct ExampleCoreAudioContext {
+    TVIAudioDeviceContext deviceContext;
+    size_t expectedFramesPerBuffer;
+    size_t maxFramesPerBuffer;
+} ExampleCoreAudioContext;
+
+// The RemoteIO audio unit uses bus 0 for ouptut, and bus 1 for input.
+static int kOutputBus = 0;
+static int kInputBus = 1;
+// This is the maximum slice size for RemoteIO (as observed in the field). We will double check at initialization time.
+static size_t kMaximumFramesPerBuffer = 1156;
+
+@interface ExampleCoreAudioDevice()
+
+@property (nonatomic, assign, getter=isInterrupted) BOOL interrupted;
+@property (nonatomic, assign) AudioUnit audioUnit;
+
+@property (nonatomic, strong, nullable) TVIAudioFormat *renderingFormat;
```

Comment:
We don't use `nullable` for private properties

Reply:
I figured that nullability was important given that this code is hosted inside a Swift app. Lets discuss.